packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -299,6 +299,14 @@ - (void)handleCreateMethodCall:(FlutterMethodCall *)call
   });
 }
 
+// Returns integer value if provided and positive, or nil if not specified or negative.
+NSNumber *toPositiveOrNull(NSNumber *number) {
+  int value = (!number || [number isEqual:[NSNull null]] || isnan([number doubleValue]))
```

Comment:
If the number is expected to be an integer, why are you calling `doubleValue` on it?

Reply:
Ok, `number` incoming from flutter method channel is `id` actually...
It can be `nil`, `NSNull`, `NSDecimalNumber`.
Also let be less restrictive and allow non-integral numbers, so it can be `NSNumber` holding `double` or `float`, including `nan`, so checking `doubleValue` for `nan` here.
Finally, rewritten as:
```dart
NSNumber *toPositiveOrNil(id number) {
  int value = (!number ||
               [number isEqual:[NSNull null]] ||
               ![number isKindOfClass:[NSNumber class]] ||
               isnan([number doubleValue]))
                  ? 0
                  : [number intValue];
  return value > 0 ? @(value) : nil;
}
```